### PR TITLE
Docs: Add examples of Apache server configurations

### PR DESCRIPTION
**Pull request by alrra**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/sonarwhal/sonarwhal)
- [x] Followed the [commit message guidelines](https://sonarwhal.com/docs/contributor-guide/contributing/pull-requests/#commitmessages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- [ ] Added/Updated related tests.

## Short description of the change(s)
  
(Partially) Fix: #649

### CODE REVIEW

File: docs/user-guide/rules/highest-available-document-mode.md [hunk 1]
```diff
@@ -216,6 +216,73 @@ X-UA-Compatible: ie=edge
...
+    <FilesMatch "\.(appcache|atom|bbaw|bmp|crx|css|cur|eot|f4[abpv]|flv|geojson|gif|htc|ic[os]|jpe?g|m?js|json(ld)?|m4[av]|manifest|map|markdown|md|mp4|oex|og[agv]|opus|otf|pdf|png|rdf|rss|safariextz|svgz?|swf|topojson|tt[cf]|txt|vcard|vcf|vtt|webapp|web[mp]|webmanifest|woff2?|xloc|xml|xpi)$">
+        Header unset X-UA-Compatible
+    </FilesMatch>
+
+</IfModule>
+```
+
+## Removing the `X-UA-Compatible` header
+
+If the header is send, in most cases, to make Apache stop sending
```
Comment (sarvaje) [inline docs/user-guide/rules/highest-available-document-mode.md]: `If the header is sent`

Comment (alrra) [reply docs/user-guide/rules/highest-available-document-mode.md]: @sarvaje Fixed, thanks!

File: docs/user-guide/rules/content-type.md [hunk 2]
```diff
@@ -106,6 +106,182 @@ HTTP/... 200 OK
...
+```
+
+The same goes for mapping certain filename extensions to specific
+charsets, which can be done using the [`AddDefaultCharset`][adddefaultcharset]
+and [`AddCharset`][addcharset] directives.
+
+If you don't want to start from scratch, below is a generic starter
+snippet that contains the necessary mappings to ensure that commonly
+used file types are served with the appropriate `Content-Type` response
+header, and thus, make your web site/app passes this rule.
```
Comment (molant) [inline docs/user-guide/rules/content-type.md]: > pass

File: docs/user-guide/rules/http-cache.md [hunk 3]
```diff
@@ -190,6 +149,176 @@ Cache-Control: no-cache
...
+
+`Cache-Control` header's `max-age` values can be set using the
+[`ExpiresDefault`][expiresdefault] and [`ExpiresByType`][expiresbytype]
+directives. Other values such as `immutable` can be set using the
+[`Header`][header directive] directive.
+
+If you don't want to start from scratch, below is a generic starter
+snippet that contains the necessary configurations to ensure that
+commonly used file types are served with the appropriate `Cache-Control`
+header, and thus, make your web site/app passes this rule.
```
Comment (molant) [inline docs/user-guide/rules/http-cache.md]: > pass

File: docs/user-guide/rules/http-cache.md [hunk 4]
```diff
@@ -190,6 +149,176 @@ Cache-Control: no-cache
...
+
+Important notes:
+
+* Do not used the following snippet if you are not doing filename revin.
```
Comment (molant) [inline docs/user-guide/rules/http-cache.md]: > Do not use

> filename revving

File: docs/user-guide/rules/http-cache.md [hunk 5]
```diff
@@ -190,6 +149,176 @@ Cache-Control: no-cache
...
+* The following relies on Apache being configure to have the correct
```
Comment (molant) [inline docs/user-guide/rules/http-cache.md]: > configured

File: docs/user-guide/rules/content-type.md (hunk 2 above)
Comment (alrra) [reply docs/user-guide/rules/content-type.md]: Fixed.

File: docs/user-guide/rules/http-cache.md (hunk 3 above)
Comment (alrra) [reply docs/user-guide/rules/http-cache.md]: Fixed.

File: docs/user-guide/rules/http-cache.md (hunk 4 above)
Comment (alrra) [reply docs/user-guide/rules/http-cache.md]: Fixed.

File: docs/user-guide/rules/http-cache.md (hunk 5 above)
Comment (alrra) [reply docs/user-guide/rules/http-cache.md]: Fixed.

**Pull request closed without merging**